rofl-app-core/src/registration.rs

```diff
@@ -80,20 +78,39 @@ where
     async fn run(mut self) {
         slog::info!(self.logger, "starting registration task");
 
-        // TODO: Handle retries etc.
+        let mut last_registration_epoch: Option<EpochTime> = None;
+
         while self.notify.recv().await.is_some() {
-            if let Err(err) = self.refresh_registration().await {
-                slog::error!(self.logger, "failed to refresh registration";
-                    "err" => ?err,
-                );
+            let backoff = backoff::ExponentialBackoff::default();
+
+            let result = backoff::future::retry(backoff, async || {
+                let result = self.refresh_registration(last_registration_epoch).await;
+                if let Err(ref err) = result {
+                    slog::error!(self.logger, "failed to refresh registration";
+                        "err" => ?err,
+                    );
+                }
+
+                result.map_err(backoff::Error::transient)
+            })
+            .await;
+
+            match result {
+                Ok(epoch) => last_registration_epoch = Some(epoch),
```

Comment:
Do we need to clear the `notify` buffer after successful registration, as otherwise we will register twice in a row?

Reply:
Why, the `refresh_registration` method should check before registering again?

Reply:
Did know that was implemented in `refresh_registration`. Looks fine this way.